DB/src/main/java/io/deephaven/db/tables/utils/TableManagementTools.java

```diff
@@ -45,98 +51,159 @@ private TableManagementTools() {
-        final String description = "Stand-alone V2 table from " + path;
         final TableLocationProvider locationProvider = new ReadOnlyLocalTableLocationProvider(
                 StandaloneTableKey.getInstance(),
-                new StandaloneLocalTableLocationScanner(path),
+                new StandaloneLocalTableLocationScanner(location),
                 false,
                 TableDataRefreshService.getSharedRefreshService());
-        return new SimpleSourceTable(tableDefinition, description, RegionedTableComponentFactoryImpl.INSTANCE, locationProvider, null);
+        return getTable("Stand-alone V2 table from " + path, tableDefinition, locationProvider);
+    }
+
+    private static Table readTableFromSingleParquetFile(@NotNull final File sourceFile, @NotNull final TableDefinition tableDefinition) {
+        final TableLocationProvider locationProvider = new ReadOnlyLocalTableLocationProviderByParquetFile(
+                StandaloneTableKey.getInstance(),
+                sourceFile,
+                false,
+                TableDataRefreshService.getSharedRefreshService());
+        return getTable("Read single parquet file from " + sourceFile, tableDefinition, locationProvider);
     }
 
     /**
      * Reads in a table from disk.
      *
-     * @param path table location
+     * @param sourceFilePath table location; if it ends in ".parquet" is assumed to be a single file location, otherwise is a directory.
      * @return table
      */
-    public static Table readTable(@NotNull final File path) {
-        final TableDefinition tableDefinition = TableDefinition.loadDefinition(path, path.getName());
-        if (tableDefinition == null) {
-            throw new IllegalStateException(path + " doesn't have a loadable TableDefinition");
+    public static Table readTable(@NotNull final String sourceFilePath) {
+        return readParquetTable(new File(sourceFilePath), !sourceFilePath.endsWith(PARQUET_FILE_EXTENSION));
+    }
+
+    /**
+     * Reads in a table from disk.
+     *
+     * @param sourceFilePath table location; if its path ends in ".parquet" is assumed to be a single file location, otherwise is a directory.
+     * @return table
+     */
+    public static Table readTable(@NotNull final File sourceFilePath) {
+        return readParquetTable(sourceFilePath, !sourceFilePath.getPath().endsWith(PARQUET_FILE_EXTENSION));
+    }
+
+    private static Table readParquetTable(@NotNull final File source, final boolean isDirectory) {
+        final ArrayList<ColumnDefinition> cols = new ArrayList<>();
+        final ParquetReaderUtil.ColumnDefinitionConsumer colConsumer =
+                (final String name, final Class<?> dbType, Class<?> componentType, final boolean isGrouping,
+                 final String codecName, final String codecArgs, final int objectWidth) -> {
+                    final ColumnDefinition<?> colDef;
+                    if (codecName != null) {
+                        if (objectWidth != -1) {
```

Comment:
```
                        final ObjectCodec<?> codec = CodecCache.DEFAULT.getCodec(codecName, codecArgs);
                        if (codec.expectedObjectWidth() != ObjectDecoder.VARIABLE_WIDTH_SENTINEL) {
```

Reply:
Done, thanks.   Much better not to have yet another metadata annotation in the poor file.